src/NServiceBus.Core/Features/EndpointConfigurationExtensions.cs

```diff
@@ -53,5 +58,17 @@ public static void DisableFeature(this EndpointConfiguration config, Type featur
 
             config.Settings.DisableFeature(featureType);
         }
+
+        internal static IEnumerable<Type> GetExplicitlyEnabledFeatures(this SettingsHolder settings)
```

Comment:
Is there any impact on the feature order or the usage of the settings extensions such as https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/master/src/NServiceBus.Persistence.CosmosDB/Saga/SagaStorage.cs#L14 that we might need to take into account by doing this?

Reply:
I tried to keep the order the same way as is in default config by using the scanned type first and then adding all manually registered types. Not sure how this might impact the order when not using scanning at all. But the type detection is completely decoupled from the settings extensions that control the feature's enabled state (which is tracked in the settings). Do you have a specific scenario in mind that I could test for potential problems?